src/internal/carriers/rates/rate.ts

```diff
@@ -34,11 +34,13 @@ export class Rate {
   public readonly charges: readonly Charge[];
   public readonly totalAmount: MonetaryValue;
   public readonly notes: readonly Note[];
-  public readonly packages: readonly RatePackage[];
+  public readonly packages: RatePackage[];
   public readonly deliveryConfirmation?: DeliveryConfirmation;
 
-  public get package(): RatePackage {
-    return this.packages[0];
+  public get package(): RatePackage | undefined {
+    if(this.packages && this.packages.length > 0) {
+      return this.packages[0];
+    }
   }
```

Comment:
The `if` statement is unnecessary, since `this.packages` will always be an array.  All that's needed is `return this.packages[0];`